CRM/Core/Payment/Elavon.php

```diff
@@ -231,14 +230,13 @@ public function doPayment(&$params, $component = 'contribute') {
       // Success !
       if ($this->_mode !== 'test') {
         // 'trxn_id' is varchar(255) field. returned value is length 37
-        $params['trxn_id'] = $processorResponse['ssl_txn_id'];
+        $result['trxn_id'] = $processorResponse['ssl_txn_id'];
       }
 
       $params['trxn_result_code'] = $processorResponse['ssl_approval_code'] . "-Cvv2:" . $processorResponse['ssl_cvv2_response'] . "-avs:" . $processorResponse['ssl_avs_response'];
-      $params['payment_status_id'] = array_search('Completed', $statuses);
```

Comment:
@demeritcowboy I realised this PR didn't have the linked lab issue and previous PR - now added to the description. We can't actually change the pass-by-reference signature because that will cause PHP errors but basically no calling code should be using the pass-by-reference result (none is in core and I'm not aware of third-party code that is).
`trxn_result_code` was set by various core processors but is not used anywhere - maybe it was once upon a time.